### PR TITLE
Chore/serde restructure

### PR DESCRIPTION
**Pull request by arujit**
- Comparing the producer-consumer API changes with the base code restructure branch.

### CODE REVIEW

File: dagger-common/build.gradle [hunk 1]
```diff
@@ -60,9 +60,11 @@ dependencies {
     compileOnly group: 'org.apache.flink', name: 'flink-connector-kafka_2.11', version: flinkVersion
 
     dependenciesCommonJar 'org.apache.flink:flink-metrics-dropwizard:' + flinkVersion
+    dependenciesCommonJar 'org.apache.flink:flink-json:' + flinkVersion
+    dependenciesCommonJar 'com.jayway.jsonpath:json-path:2.4.0'
     dependenciesCommonJar 'com.gojek:stencil:2.0.15'
 
-    testImplementation 'junit:junit:4.12'
+    testImplementation 'junit:junit:4.13'
```
Comment (arujit) [inline dagger-common/build.gradle:67]: This is to keep the Junit version same as the core.

File: dagger-common/src/main/java/io/odpf/dagger/common/protohandler/TimestampProtoHandler.java [hunk 2]
```diff
@@ -44,6 +46,11 @@ public boolean canHandle() {
         if (field instanceof java.sql.Timestamp) {
             timestamp = convertSqlTimestamp((java.sql.Timestamp) field);
         }
+
```
[the remark this answers is not in the document]
Comment (prakharmathur82) [reply dagger-common/src/main/java/io/odpf/dagger/common/protohandler/TimestampProtoHandler.java:49]: didn't get why this is required, also if this is required can you add the corresponding tests?

File: dagger-core/build.gradle [hunk 3]
```diff
@@ -118,7 +118,7 @@ jacocoTestCoverageVerification {
     violationRules {
         rule {
             limit {
-                minimum = 0.89
+                minimum = 0.88
```
Comment (prakharmathur82) [inline dagger-core/build.gradle:121]: can we do something about it?

File: dagger-core/src/main/java/io/odpf/dagger/core/Streams.java [hunk 4]
```diff
@@ -46,32 +48,28 @@
...
     /**
-     * Gets streams.
+     * Gets kafkaSource.
      *
-     * @return the streams
+     * @return the Sources
      */
-    public Map<String, FlinkKafkaConsumerCustom> getStreams() {
-        return streams;
+    public Map<String, KafkaSource> getKafkaSource() {
```
Comment (prakharmathur82) [inline dagger-core/src/main/java/io/odpf/dagger/core/Streams.java:71]: should be getKafkaSources

File: dagger-core/src/main/java/io/odpf/dagger/core/sink/ProtoSerializer.java [hunk 5]
```diff
@@ -55,15 +55,27 @@ public ProtoSerializer(String keyProtoClassName, String messageProtoClassName, S
      * @param outputTopic               the output topic
      */
     public ProtoSerializer(String keyProtoClassName, String messageProtoClassName, String[] columnNames, StencilClientOrchestrator stencilClientOrchestrator, String outputTopic) {
-        this(keyProtoClassName, messageProtoClassName, columnNames, stencilClientOrchestrator);
```
Comment (prakharmathur82) [inline dagger-core/src/main/java/io/odpf/dagger/core/sink/ProtoSerializer.java:58]: this was doing the same thing

File: dagger-core/src/main/java/io/odpf/dagger/core/metrics/reporters/ErrorReporterFactory.java [hunk 6]
```diff
@@ -38,4 +45,12 @@ public static ErrorReporter getErrorReporter(RuntimeContext runtimeContext, Bool
             return new NoOpErrorReporter();
         }
     }
+
+    public static ErrorReporter getErrorReporter(MetricGroup metricGroup, Boolean telemetryEnable, long shutDownPeriod) {
```
Comment (prakharmathur82) [inline dagger-core/src/main/java/io/odpf/dagger/core/metrics/reporters/ErrorReporterFactory.java:49]: use this only

**Pull request merged**